samples/python/qrcode.py

```diff
@@ -0,0 +1,157 @@
+        if isinstance(decode_info, str):
+            decode_info = [decode_info]
+        if n > 0:
+            for i in range(n):
+                cnt = np.array(points[i]).reshape((-1, 1, 2)).astype(np.int32)
+                self.drawQRCodeContours(result, cnt)
+                msg = 'QR[{:d}]@{} :'.format(i, *(cnt.reshape(1, -1).tolist()))
+                print (msg, end="")
+                if len(decode_info) > i:
+                    if decode_info[i]:
+                        print("'", decode_info[i], "'")
+                    else:
+                        print("Can't decode QR code")
+                else:
+                    print("Decode information is not available (disabled)")
+        else:
+            print ("QRCode not  detected!")
+        self.drawFPS(result, fps)
+
+    def runQR(self, qrCode, inputimg):
+        if not self.multi:
+            if not self.detect:
+                decode_info, points, _ = qrCode.detectAndDecode(inputimg)
+                dec_info = decode_info
+            else:
+                _, points = qrCode.detect(inputimg)
+                dec_info = []
+        else:
+            if not self.detect:
+                _, decode_info, points, _ = qrCode.detectAndDecodeMulti(
+                    inputimg)
+                dec_info = decode_info
+            else:
+                _, points = qrCode.detectMulti(inputimg)
+                dec_info = []
+        if points is None:
+            points = []
+        return points, dec_info
+
+    def DetectQRFrmImage(self, inputfile):
+        inputimg = cv.imread(inputfile, cv.IMREAD_COLOR)
+        qrCode = cv.QRCodeDetector()
+        count = 10
+        timer = cv.TickMeter()
+        for _ in range(count):
+            timer.start()
+            points, decode_info = self.runQR(qrCode, inputimg)
+            timer.stop()
+        fps = count / timer.getTimeSec()
+        print("FPS: ", fps)
+        result = inputimg
+        self.drawQRCodeResults(result, points, decode_info, fps)
+        cv.imshow("QR", result)
+        cv.waitKey(1)
+        if self.out != '':
+            outfile = self.fname + self.fext
+            print ("Saving Result: ", outfile)
+            cv.imwrite(outfile, result)
+
+        print("Press any key to exit ...")
+        cv.waitKey(0)
+        print("Exit")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='This program detects the QR-codes input images using OpenCV Library.')
+    parser.add_argument(
+        '-i',
+        '--input',
+        help="input image path (default input file path is 'opencv_extra/testdata/cv/qrcode/multiple/*_qrcodes.png",
+        default="../../../opencv_extra/testdata/cv/qrcode/multiple/*_qrcodes.png",
+        metavar="")
+    parser.add_argument(
+        '-d',
+        '--detect',
+        help="detect QR code only (skip decoding) (default value is False)",
+        action='store_true')
+    parser.add_argument(
+        '-m',
+        '--multi',
```

Comment:
I propose to run detectMulti by default and inverse the option like for detection case. The QR code images in OpenCV extra contains several QR codes per image and the sample started with default options produces garbage.